x-pack/plugins/observability_solution/uptime/server/legacy_uptime/lib/alerts/common.ts

```diff
@@ -115,17 +116,21 @@ export const setRecoveredAlertsContext = async ({
       basePath.publicBaseUrl
     );
 
-    alert.setContext({
-      ...state,
-      [ALERT_DETAILS_URL]: alertUrl,
+    alertsClient!.setAlertData({
+      id: recoveredAlertId,
+      context: {
+        ...state,
+        [ALERT_DETAILS_URL]: alertUrl,
+      },
     });
   }
 };
 
 export const uptimeRuleTypeFieldMap = { ...uptimeRuleFieldMap, ...legacyExperimentalFieldMap };
 
-export const UptimeRuleTypeAlertDefinition: IRuleTypeAlerts = {
+export const UptimeRuleTypeAlertDefinition: IRuleTypeAlerts<ObservabilityUptimeAlert> = {
```

Comment:
There's another definition for this in `x-pack/plugins/observability_solution/synthetics/server/alert_rules/common.ts`. Should we remove one definition?

Reply:
I think they have different usages.
This one is for the legacy, duration, status check and tls.
The one you shared is for synthetics monitor_status and tls.
I can remove one of them when the synthetic rules are onboarded as well. Can't use `shouldWrite: true` on them until then.